api/wing/schema.py

```diff
@@ -73,12 +74,37 @@ def mutate(self, info, wing_id, **kwargs):
             return UpdateWing(wing=exact_wing)
 
 
-class Query(graphene.ObjectType):
-    all_wings = graphene.List(Wing)
+class PaginatedWings(Paginate):
+    wings = graphene.List(Wing)
 
-    def resolve_all_wings(self, info):
+    def resolve_wings(self, info, **kwargs):
+        ''' This function paginates the returned response
+        if page parameter is passed,
+        otherwise it returns all floors
+        '''
+        page = self.page
+        per_page = self.per_page
         query = Wing.get_query(info)
-        return query.all()
+        if not page:
+            return query.all()
+        page = validate_page(page)
+        self.query_total = query.count()
+        result = query.limit(per_page).offset(page*per_page)
+        if result.count() == 0:
+            return GraphQLError("No more resources")
```

Comment:
Can you use a more descriptive message here? something like `No wing on this page` or something better, considering we also have resources schema in the application, this error message can be confusing